Sources/TuistKit/Services/DocService.swift

```diff
@@ -0,0 +1,127 @@
+import Foundation
+import RxBlocking
+import Signals
+import TSCBasic
+import TuistCore
+import TuistDoc
+import TuistSupport
+
+// MARK: - DocServicing
+
+protocol DocServicing {
+    func run(path: AbsolutePath) throws
+}
+
+// MARK: - DocService
+
+struct DocService {
+    private let projectGenerator: ProjectGenerating
+    private let swiftDocController: SwiftDocControlling
+    private let swiftDocServer: SwiftDocServing
+    private let fileHandler: FileHandling
+
+    init(projectGenerator: ProjectGenerating = ProjectGenerator(),
+         swiftDocController: SwiftDocControlling = SwiftDocController(),
+         swiftDocServer: SwiftDocServing = SwiftDocServer(),
+         fileHandler: FileHandling = FileHandler())
+    {
+        self.projectGenerator = projectGenerator
+        self.swiftDocController = swiftDocController
+        self.swiftDocServer = swiftDocServer
+        self.fileHandler = fileHandler
+    }
+
+    func run(project path: AbsolutePath, target targetName: String, serve: Bool, port: UInt16) throws {
+        let (_, graph, _) = try projectGenerator.loadProject(path: path)
+
+        let targets = graph.targets(at: path)
+            .filter { !$0.dependsOnXCTest }
+            .map { $0.target }
+
+        guard let target = targets.first(where: { $0.name == targetName }) else {
+            throw DocService.Error.targetNotFound(name: targetName)
+        }
+
+        let sources = target.sources.map(\.path)
+
+        try withTemporaryDirectory { generationDirectory in
+            let parameters = Parameters(serve: serve,
+                                        swiftDocServer: swiftDocServer,
+                                        port: port,
+                                        generationDirectory: generationDirectory.pathString)
+
+            try swiftDocController.generate(
+                format: parameters.format,
+                moduleName: targetName,
+                baseURL: parameters.baseURL,
+                outputDirectory: generationDirectory.pathString,
+                sourcesPaths: sources
+            )
+
+            let indexPath = generationDirectory.appending(component: parameters.indexName)
+
+            guard fileHandler.exists(indexPath) else {
+                throw Error.documentationNotGenerated
+            }
+
+            if serve {
+                try swiftDocServer.serve(path: generationDirectory, port: port)
+            } else {
+                logger.pretty("You can find the documentation at \(generationDirectory.pathString)")
+            }
+        }
+    }
+}
+
+// MARK: - Parameters
+
+extension DocService {
```

Comment:
To be consistent with the rest of the codebase, and avoid passing unnecessary complexity to the compiler, I'd keep everything at the root of the file:

```swift
struct DocServiceParameters {}
enum DocServiceError: Error {}
```

I'd place both of them at the top of the file.